mantidimaging/gui/windows/spectrum_viewer/presenter.py

```diff
@@ -312,3 +344,25 @@ def do_remove_roi(self, roi_name: str | None = None) -> None:
     def handle_export_tab_change(self, index: int) -> None:
         self.export_mode = ExportMode(index)
         self.view.on_visibility_change()
+
+    def handle_tof_unit_change(self) -> None:
+        selected_mode = self.view.tof_mode_select_group.checkedAction().text()
+        self.model.tof_mode = self.view.allowed_modes[selected_mode]
+        self.model.set_relevant_tof_units()
+        tof_mode = self.model.tof_mode
+        tof_axis_label = ""
+        if tof_mode == ToFUnitMode.IMAGE_NUMBER:
+            tof_axis_label = "Image index"
+        if tof_mode == ToFUnitMode.TOF_US:
+            tof_axis_label = "Time of Flight (\u03BC s)"
+        if tof_mode == ToFUnitMode.WAVELENGTH:
+            tof_axis_label = "Neutron Wavelength (\u212B)"
+        if tof_mode == ToFUnitMode.ENERGY:
+            tof_axis_label = "Neutron Energy (MeV)"
```

Comment:
Can these strings be put in a dictionary, so this looks more like
`tof_axis_label = UNIT_NAMES[self.model.tof_mode]`

There is already the `allowed_modes` dict. Maybe that could have the full strings with units. Then when making the label you could use `.partition(' (')[0]` to remove the unit.

Reply:
I've changed it such that the `allowed modes` dict now looks like 
```
self.allowed_modes = {
            "Image Index": [ToFUnitMode.IMAGE_NUMBER, "Image index"],
            "Wavelength": [ToFUnitMode.WAVELENGTH, "Neutron Wavelength (\u212B)"],
            "Energy": [ToFUnitMode.ENERGY, "Neutron Energy (MeV)"],
            "us": [ToFUnitMode.TOF_US, "Time of Flight (\u03BC s)"]
        }
```

so the tof label is now handled as: 

```
selected_mode = self.view.tof_mode_select_group.checkedAction().text()
        self.model.tof_mode = self.view.allowed_modes[selected_mode][0]
        self.model.set_relevant_tof_units()
        self.view.spectrum_widget.spectrum_plot_widget.set_tof_axis_label(self.view.allowed_modes[selected_mode][1])
```

which is a lot more streamlined!